polygeist/include/polygeist/Passes/Utils.h

```diff
@@ -107,3 +107,15 @@ static inline bool hasElse(mlir::scf::IfOp op) {
 static inline bool hasElse(mlir::AffineIfOp op) {
   return op.getElseRegion().getBlocks().size() > 0;
 }
+
+/// States whether a MemRefType can be lowered to a bare pointer
+///
+/// Only ranked MemRefTypes with identity map and non-dynamic dimensions in the
+/// range [1, rank) can be lowered to a bare pointer.
+inline bool canBeLoweredToBarePtr(mlir::MemRefType memRefType) {
+  if (!memRefType.getLayout().isIdentity() || !memRefType.hasRank())
+    return false;
+  const auto shape = memRefType.getShape();
+  return std::none_of(shape.begin() + 1, shape.end(),
+                      mlir::ShapedType::isDynamic);
```

Comment:
I see there are many MLIR tests to check that memrefs are lowered to a llvm.ptr (in bareptrlowering.mlir). 
Just checking, do we have tests to check that memrefs that *cannot* be converted to a llvm.ptr (cases for which this function would return false) are not lowered to a llvm,ptr ?

Reply:
> Just checking, do we have tests to check that memrefs that _cannot_ be converted to a llvm.ptr (cases for which this function would return false) are not lowered to a llvm,ptr ?

Everything's working fine for non-identity layout maps, but the upstream case is failing in the dynamic dimensions case. I have a couple of solutions in mind, but, as cgeist won't generate this kind of memrefs, I think we can tackle this in a different PR.

Reply:
> Everything's working fine for non-identity layout maps

We're being conservative here. We could be lowering these, but, as this would require rethinking most patterns and no kernel generated by cgeist will present this kind of memrefs, we can just skip this for now.

Reply:
I think is fine to let the upstream lowering happen for this subcase. If we ever need to fix it we we can do that later.